### PR TITLE
Remove channel scoping, add featured in... for first 3 channels

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Gets rid of any residual scoping by channel, and adds a rudimentary top level 'channels' list on the top of the (now named) channels page.

Creates three 'featured in...' sections for the first three channels on the device.

## Issues addressed

Partly addresses #2051 

image

image

### CODE REVIEW

File: kolibri/plugins/learn/assets/src/state/actions/main.js [hunk 1]
```diff
@@ -294,13 +267,14 @@ function showTopicsContent(store, channelId, id) {
   ]).only(
     samePageCheckGenerator(store),
     ([content, channels, nextContent, ancestors]) => {
-      const currentChannel = getCurrentChannelObject(store.state);
+      const currentChannel = getChannelObject(store.state, channelId);
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/state/actions/main.js]: `channelId` is undefined.

File: kolibri/plugins/learn/assets/src/state/actions/main.js [hunk 2]
```diff
@@ -409,30 +362,23 @@ function showExamList(store, channelId) {
...
+    samePageCheckGenerator(store),
+    exams => {
+      const pageState = {};
+      pageState.exams = exams.map(_examState);
+      store.dispatch('SET_PAGE_STATE', pageState);
+      store.dispatch('CORE_SET_PAGE_LOADING', false);
+      store.dispatch('CORE_SET_ERROR', null);
+      store.dispatch(
+        'CORE_SET_TITLE',
+        translator.$tr('examsForChannelPageTitle', { currentChannelTitle: currentChannel.title })
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/state/actions/main.js]: `currentChannel` is undefined.

File: kolibri/plugins/learn/assets/src/state/actions/main.js [hunk 3]
```diff
@@ -361,44 +335,23 @@ function showContentUnavailable(store) {
   store.dispatch('CORE_SET_TITLE', translator.$tr('contentUnavailablePageTitle'));
 }
 
-function redirectToChannelSearch(store) {
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/state/actions/main.js:364]: Also remove from exports

File: kolibri/core/assets/src/state/getters.js [hunk 4]
```diff
@@ -37,32 +37,15 @@ function facilityConfig(state) {
...
- * Returns the 'default' channel ID:
- * - if there are channels and they match the cookie, return that
- * - else if there are channels, return the first one
- * - else return null
- *
- * Not truly a 'getter' because it doesn't use vuex state
- */
-function getDefaultChannelId(channelList) {
-  if (channelList && channelList.length) {
-    const cookieVal = cookiejs.get('currentChannelId');
```
Comment (christianmemije) [inline kolibri/core/assets/src/state/getters.js:50]: Removed unused `cookiejs` import

File: kolibri/core/assets/src/state/actions.js [hunk 5]
```diff
@@ -367,33 +366,10 @@ function initContentSession(store, channelId, contentId, contentKind) {
...
 }
 
-/*
- * Set channel state info.
- */
-function _setChannelState(store, currentChannelId, channelList) {
-  store.dispatch('SET_CORE_CHANNEL_LIST', channelList);
-  store.dispatch('SET_CORE_CURRENT_CHANNEL', currentChannelId);
-  if (currentChannelId) {
-    cookiejs.set('currentChannelId', currentChannelId);
```
Comment (christianmemije) [inline kolibri/core/assets/src/state/actions.js:377]: Removed unused `cookiejs` import

**Comment by christianmemije** [comment]
There are still 2 references to `state.store.channels.list` that can be replaced with `getChannels` in `wizard-export` and `import-preview`.

File: kolibri/plugins/learn/assets/src/constants.js [hunk 6]
```diff
@@ -1,11 +1,10 @@
 // a name for every URL pattern
 const PageNames = {
-  TOPICS_ROOT: 'TOPICS_ROOT',
+  CHANNELS: 'CHANNELS',
   TOPICS_CHANNEL: 'TOPICS_CHANNEL',
   TOPICS_TOPIC: 'TOPICS_TOPIC',
   TOPICS_CONTENT: 'TOPICS_CONTENT',
-  LEARN_ROOT: 'LEARN_ROOT',
-  LEARN_CHANNEL: 'LEARN_CHANNEL',
+  LEARN_RECOMMENDED: 'LEARN_RECOMMENDED',
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/constants.js]: Remove the LEARN prefix

**Comment by rtibbles** [comment]
@christianmemije I didn't remove those references on purpose, as they would conflict with @jonboiser's work.

File: kolibri/plugins/learn/assets/src/constants.js [hunk 7]
```diff
@@ -1,11 +1,10 @@
 // a name for every URL pattern
 const PageNames = {
-  TOPICS_ROOT: 'TOPICS_ROOT',
+  CHANNELS: 'CHANNELS',
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/constants.js]: I'd prefer if the pageName mapped nicely with the URL.

File: kolibri/plugins/learn/assets/src/constants.js [hunk 8]
```diff
@@ -1,11 +1,10 @@
...
   TOPICS_CHANNEL: 'TOPICS_CHANNEL',
   TOPICS_TOPIC: 'TOPICS_TOPIC',
   TOPICS_CONTENT: 'TOPICS_CONTENT',
-  LEARN_ROOT: 'LEARN_ROOT',
-  LEARN_CHANNEL: 'LEARN_CHANNEL',
+  LEARN_RECOMMENDED: 'LEARN_RECOMMENDED',
   RECOMMENDED_POPULAR: 'RECOMMENDED_POPULAR',
   RECOMMENDED_RESUME: 'RECOMMENDED_RESUME',
   RECOMMENDED_NEXT_STEPS: 'RECOMMENDED_NEXT_STEPS',
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/constants.js]: `LEARN_CONTENT` -> `RECOMMENDED_CONTENT`
`SEARCH_ROOT` is no longer used.
`EXAM_ROOT`?

File: kolibri/plugins/learn/assets/src/constants.js (hunk 7 above)
Comment (rtibbles) [reply kolibri/plugins/learn/assets/src/constants.js]: So, change all TOPICS references to CHANNELS? (including the URL?)

File: kolibri/plugins/learn/assets/src/constants.js (hunk 8 above)
Comment (rtibbles) [reply kolibri/plugins/learn/assets/src/constants.js]: EXAM_ROOT was removed in error, reinstating.

File: kolibri/plugins/learn/assets/src/constants.js (hunk 7 above)
Comment (christianmemije) [reply kolibri/plugins/learn/assets/src/constants.js]: As long as it's consistent. Topics makes more sense than channels, but `BROWSE` makes more sense than either... You browse content and topics...

File: kolibri/plugins/learn/assets/src/state/actions/recommended.js [hunk 9]
```diff
@@ -10,8 +10,8 @@ import {
 import {
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/state/actions/recommended.js:10]: Unused imports here and above.

File: kolibri/plugins/learn/assets/src/views/channels-page/index.vue [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    />
+
+  </div>
+
+</template>
+
+
+<script>
+
+  import { getChannels } from 'kolibri.coreVue.vuex.getters';
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/views/channels-page/index.vue]: Unused imports getChannels, ContentNodeKinds, some, forEach

File: kolibri/plugins/learn/assets/src/views/content-card-group-header/index.vue [hunk 11]
```diff
@@ -9,7 +9,7 @@
         </h2>
 
         <span v-if="showViewMore" class="card-group-details-view-more">
-          <router-link to="viewMorePageLink">
+          <router-link :to="viewMorePageLink">
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/views/content-card-group-header/index.vue:12]: That helps :sweat_smile:

File: kolibri/plugins/learn/assets/src/views/index.vue [hunk 12]
```diff
@@ -16,9 +15,9 @@
         />
         <k-navbar-link
           type="icon-and-title"
-          :title="$tr('topics')"
+          :title="$tr('channels')"
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/views/index.vue]: Leave as topics?
https://projects.invisionapp.com/share/ZTA3FH2S2#/screens/242524248

File: kolibri/plugins/learn/assets/src/views/index.vue [hunk 13]
```diff
@@ -53,13 +52,13 @@
   import { TopLevelPageNames } from 'kolibri.coreVue.vuex.constants';
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/views/index.vue:51]: `pageModes` is unused. An eslint text editor plugin comes in handy.

File: kolibri/plugins/learn/assets/src/views/learn-page/index.vue [hunk 14]
```diff
@@ -133,15 +127,26 @@
     methods: {
       genContentLink(id, kind) {
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/views/learn-page/index.vue]: `kind` is not used.

File: kolibri/plugins/learn/assets/src/views/topics-page/index.vue [hunk 15]
```diff
@@ -22,7 +22,6 @@
 
 <script>
 
-  import { getCurrentChannelObject } from 'kolibri.coreVue.vuex.getters';
   import { PageNames } from '../../constants';
   import { ContentNodeKinds } from 'kolibri.coreVue.vuex.constants';
   import some from 'lodash/some';
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/views/topics-page/index.vue]: some and forEach unused.

**Comment by christianmemije** [comment]
There should be breadcrumbs at this level

127 0 0 1-8000-learn- 1

https://projects.invisionapp.com/share/ZTA3FH2S2#/screens/226351535

**Comment by rtibbles** [comment]
Cleaned up - not a huge fan of the topics/channels inconsistency, but it's up to spec now.

**Pull request merged**